runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -225,15 +249,22 @@ def model_predict(self, df: pd.DataFrame, horizon: int = None) -> pd.DataFrame:
         result_df["ts_id"] = str(df["ts_id"].iloc[0])
         return result_df[return_cols]
 
-    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input_df: pd.DataFrame) -> pd.DataFrame:
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, model_input: pd.DataFrame) -> pd.Series:
         """
         Predict API from mlflow.pyfunc.PythonModel
         :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
                         can use to perform inference.
-        :param input_df: Input dataframe
+        :param model_input: Input dataframe
         :return: A pd.DataFrame with the forecast components.
         """
-        return self._predict_impl(input_df)
+        self._validate_cols(model_input, self._id_cols + [self._time_col])
+        test_df = model_input.copy()
+        test_df["ts_id"] = test_df[self._id_cols].agg('-'.join, axis=1)
+        test_df.rename(columns={self._time_col: "ds"}, inplace=True)
+        group_cols = ["ts_id"] + self._id_cols
+        predict_df = test_df.groupby(group_cols).apply(lambda df: self.model(df.name[0]).predict(df)).reset_index()
+        return_df = test_df.merge(predict_df, how="left", on=self._id_cols)
```

Comment:
wait, why do we merge only on id_cols? I think we should also merge on time_col. For example if we have a dataframe with `time_col = [pd.TimeSeries(2018, 10, 1), pd.TimeSeries(2018, 10, 2), pd.TimeSeries(2018, 10, 1), pd.TimeSeries(2018, 10, 2)]` and `id_col = [1, 1, 2, 2]`, merging only on id_cols will cause problems.

Reply:
I believe merging `id_cols` is the right thing, as it will put `[pd.Timestamp(2018, 10, 1), pd.Timestamp(2018, 10, 2)]` into a `df` and pass it into the lambda. We should add a test case to verify the behavior though.

Reply:
Could you explain? Say we have
```
test_pd = pd.DataFrame({"ds": [pd.Timestamp(2018, 10, 1), pd.Timestamp(2018, 10, 2), pd.Timestamp(2018, 10, 1), pd.Timestamp(2018, 10, 2)], "id": [1, 1, 2, 2]})
predict_df = pd.DataFrame({"ds": [pd.Timestamp(2018, 10, 1), pd.Timestamp(2018, 10, 2), pd.Timestamp(2018, 10, 1), pd.Timestamp(2018, 10, 2)], "id": [1, 1, 2, 2], "yhat": [1, 2, 3, 4]})
```
Then running `test_df.merge(predict_df, how="left", on=["id",])` returns a dataframe with 8 rows.
Did I get something wrong?

Reply:
Ah, you're right. This merge in L266 should also include the `ds` column. I though we were talking about the `groupby` in L265. Either way, having a unit test would be good @lu-wang-dl :)

Reply:
Good point. I will update it.